Bug in CMAKE leads to FMU export with missing dependency

From the discussion in #11604, it seems that there is a CMAKE bug that leads to missing dependencies in FMUs if using clang (not if using gcc). 

@bilderbuchi [notes](https://github.com/OpenModelica/OpenModelica/issues/11604#issuecomment-1975883356):

> [This comment](https://gitlab.kitware.com/cmake/cmake/-/issues/20753#note_1377270) in the cmake issue tracker sounds quite relevant, and could be the root cause, and an easy fix (tabs vs spaces). I guess that should be tracked a separate cmake issue, but I don't have an account there. Anybody?
> Can we work around that somehow in the meantime, e.g. with some sed invocation in-between, or somesuch?

I would suggest to implement his proposed workarund and, if possible, to open a ticket on the CMAKE tracker addressing this issue.

The nightly build shows this error:

```
CMake Error at cmake_install.cmake:71 (file):
  file Could not resolve runtime dependencies:

    zlib.dll


make: *** [Makefile:100: install] Error 1
Post compilation process failed. Exited with code 0x00000002.
```

--- a/OMCompiler/Compiler/SimCode/SimCodeUtil.mo
+++ b/OMCompiler/Compiler/SimCode/SimCodeUtil.mo
@@ -15895,6 +15895,7 @@ protected
   end addDockerVol;
 algorithm
   (locations, libraries) := getDirectoriesForDLLsFromLinkLibs(libs);
+  locations := listAppend({Settings.getInstallationDirectoryPath() + "/lib/${CMAKE_LIBRARY_ARCHITECTURE}/omc"}, locations); // zlib
   locations := listAppend({Settings.getInstallationDirectoryPath() + "/bin"}, locations);   // pthread located in OpenModelica/bin/ on Windows
   locations := List.map(locations, addDockerVol);
   // Use target_link_directories when CMake 3.13 is available and skip the find_library part
@@ -15907,7 +15908,7 @@ algorithm
   for lib in libraries loop
     cmakecode := cmakecode + "find_library(" + lib + "\n" +
                  "             NAMES " + lib + "\n" +
-                 "             PATHS ${EXTERNAL_LIBDIRECTORIES})\n" +
+                 "             PATHS ${EXTERNAL_LIBDIRECTORIES} NO_DEFAULT_PATH)\n" +
                  "message(STATUS \"Linking ${" + lib + "}\")" + "\n" +
                  "target_link_libraries(${FMU_NAME_HASH} PRIVATE ${" + lib + "})" + "\n" +
                  "list(APPEND RUNTIME_DEPENDS ${" + lib + "})" + "\n";

--- a/OMCompiler/Compiler/runtime/systemimpl.c
+++ b/OMCompiler/Compiler/runtime/systemimpl.c
@@ -702,7 +702,7 @@ int SystemImpl__systemCall(const char* str, const char* outFile)
   if (pID == 0) { // child
     if (*outFile) {
       /* redirect stdout, stderr in the fork'ed process */
-      int fd = open(outFile, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+      int fd = open(outFile, O_RDWR | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR);
       if (fd < 0) {
         _exit(1);
       }

